rotkehlchen/chain/evm/contracts.py

```diff
@@ -146,86 +146,130 @@ class EvmContracts(Generic[T]):
+            root_dir = Path(__file__).resolve().parent.parent.parent
+            builtin_database = root_dir / 'data' / 'global.db'
+            cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
+            result = cursor.execute(
+                'SELECT contract_data.address, contract_data.chain_id, '
+                'contract_data.deployed_block, contract_abi.name, contract_abi.value FROM '
+                'clean_db.contract_data LEFT JOIN clean_db.contract_abi ON '
+                'contract_data.abi=contract_abi.id WHERE contract_data.chain_id=? AND '
+                'contract_data.address=?',
+                bindings,
+            ).fetchone()
+            cursor.execute('DETACH DATABASE "clean_db"')
+            if result is None:
+                return None
+
+        # Copy the contract to the global db
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            abi_id = GlobalDBHandler().get_or_write_abi(
+                write_cursor=write_cursor,
+                serialized_abi=result[4],
+                abi_name=result[3],
+            )
+            write_cursor.execute(
+                'INSERT INTO contract_data(address, chain_id, abi, deployed_block) '
+                'VALUES (?, ?, ?, ?)',
+                (result[0], result[1], abi_id, result[2]),
+            )
         return EvmContract(
             address=address,
-            abi=json.loads(result[0]),  # not handling json error -- assuming DB consistency
-            deployed_block=result[1] if result[1] else 0,
+            abi=json.loads(result[4]),  # not handling json error -- assuming DB consistency
+            deployed_block=result[2] if result[2] else 0,
         )
 
-    @overload
-    def contract(self: 'EvmContracts[Literal[ChainID.ETHEREUM]]', name: 'ETHEREUM_KNOWN_CONTRACTS') -> EvmContract:  # noqa: E501
-        ...
-
-    @overload
-    def contract(self: 'EvmContracts[Literal[ChainID.OPTIMISM]]', name: Literal['to', 'do', 'PICKLE_DILL']) -> EvmContract:  # noqa: E501
-        ...
-
-    def contract(self, name: str) -> EvmContract:
-        """Gets details of an evm contract from the global DB by name
+    def contract(self, address: ChecksumEvmAddress) -> EvmContract:
+        """Gets details of an evm contract from the global DB by address
 
         Missing contract is a programming error and should never happen.
         """
-        contract = self.contract_or_none(name)
-        assert contract, f'No contract data for {name} found'
+        contract = self.contract_by_address(address=address, fallback_to_packaged_db=True)
+        assert contract, f'No contract data for {address} found'
         return contract
 
-    def abi_or_none(self, name: str) -> Optional[list[dict[str, Any]]]:
-        """Gets abi of an evm contract from the abi json file
+    def abi_or_none(
+            self, name: str,
+            fallback_to_packaged_db: bool = False,
+    ) -> Optional[list[dict[str, Any]]]:
```

Comment:
One line per argument
```suggestion
            self,
            name: str,
            fallback_to_packaged_db: bool = False,
    ) -> Optional[list[dict[str, Any]]]:
```